src/test/java/de/tum/in/www1/artemis/StudentQuestionAnswerIntegrationTest.java

```diff
@@ -0,0 +1,140 @@
+package de.tum.in.www1.artemis;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.ZonedDateTime;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.ExerciseRepository;
+import de.tum.in.www1.artemis.repository.StudentQuestionAnswerRepository;
+import de.tum.in.www1.artemis.repository.StudentQuestionRepository;
+import de.tum.in.www1.artemis.service.StudentQuestionService;
+import de.tum.in.www1.artemis.util.DatabaseUtilService;
+import de.tum.in.www1.artemis.util.RequestUtilService;
+
+public class StudentQuestionAnswerIntegrationTest extends AbstractSpringIntegrationTest {
+
+    @Autowired
+    DatabaseUtilService database;
+
+    @Autowired
+    RequestUtilService request;
+
+    @Autowired
+    CourseRepository courseRepo;
+
+    @Autowired
+    ExerciseRepository exerciseRepo;
+
+    @Autowired
+    StudentQuestionRepository studentQuestionRepository;
+
+    @Autowired
+    StudentQuestionAnswerRepository studentQuestionAnswerRepository;
+
+    @Autowired
+    StudentQuestionService studentQuestionService;
+
+    @BeforeEach
+    public void initTestCase() {
+        database.addUsers(5, 5, 1);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        database.resetDatabase();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void createStudentQuestionAnswer() throws Exception {
+        StudentQuestion studentQuestion = database.createCourseWithExerciseAndStudentQuestions().get(0);
+
+        StudentQuestionAnswer studentQuestionAnswer = new StudentQuestionAnswer();
+        studentQuestionAnswer.setAuthor(database.getUserByLogin("tutor1"));
+        studentQuestionAnswer.setAnswerText("Test Answer");
+        studentQuestionAnswer.setAnswerDate(ZonedDateTime.now());
+        studentQuestionAnswer.setQuestion(studentQuestion);
+        StudentQuestionAnswer response = request.postWithResponseBody("/api/student-question-answers", studentQuestionAnswer, StudentQuestionAnswer.class, HttpStatus.CREATED);
+
+        // trying to create same studentQuestionAnswer again --> bad request
+        request.postWithResponseBody("/api/student-question-answers", response, StudentQuestionAnswer.class, HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void editStudentQuestionAnswer() throws Exception {
+        StudentQuestionAnswer studentQuestionAnswer = createStudentQuestionAnswerOnServer();
```

Comment:
Can a student edit the answer to a question? If not could you please cover this case.

Reply:
Good point! Yes, currently everyone can edit the answers. Anyway I added a testcase to cover it, since it makes sense to test with different users.

Reply:
Students should **only** be able to edit and delete their **own** answers. They should **NOT** be able to edit and delete answers of other users.

Tutors and instructors should be able to edit and delete answers of other users.

Please implement this logic in the tests and let them fail if the server logic is not implemented correctly, then we should fix it!